platform/raspberry-pi/platform_pi.c

```diff
@@ -88,6 +91,89 @@ int wifi_nvram_defaultRead(char *in,char *out)
     return 0; 
 }
 
+static inline cJSON *json_open_backhaul_cfg(void)
+{
+    FILE *fp;
+    cJSON *json = NULL;
+    size_t len;
+    char *buff;
+
+    fp = fopen(BACKHAUL_CFG_JSON, "r");
+    if (fp == NULL) {
+        wifi_hal_error_print("%s:%d: Failed (err=%d, msg=%s) to open file:%s\n", __func__, __LINE__,
+            errno, strerror(errno), BACKHAUL_CFG_JSON);
+        return json;
+    }
+    fseek(fp, 0, SEEK_END);
+    len = ftell(fp);
+    fseek(fp, 0, SEEK_SET);
+
+    buff = malloc(len);
+    if (buff == NULL) {
+        wifi_hal_error_print("%s:%d: Failed to allocate %zu bytes for json file\n", __func__,
+            __LINE__, len);
+        return NULL;
+    }
+    len = fread(buff, 1, len, fp);
+    json = cJSON_ParseWithLength(buff, len);
+    if (json == NULL) {
+        const char *const error_ptr = cJSON_GetErrorPtr();
+        wifi_hal_error_print("%s:%d: Error json file parse: %s\n", __func__, __LINE__,
+            (error_ptr ? error_ptr : "UNKNOWN"));
+    }
+    free(buff);
+    fclose(fp);
+    return json;
+}
+
+static inline int json_parse_backhaul_keypassphrase(char *backhaul_keypassphrase)
+{
+    cJSON *backhaul_keypassphrase_item;
+    cJSON *json = NULL;
+    cJSON_bool valid;
+
+    json = json_open_backhaul_cfg();
+    if (json == NULL) {
+        return -1;
+    }
+
+    backhaul_keypassphrase_item = cJSON_GetObjectItem(json, "Backhaul_KeyPassphrase");
+    if (!(valid = cJSON_IsString(backhaul_keypassphrase_item))) {
+        wifi_hal_error_print("%s:%d: (InterfaceName) does "
+                             "not exist or is not a string\n",
+            __func__, __LINE__);
+        cJSON_Delete(json);
+        return -1;
+    }
+    strncpy(backhaul_keypassphrase, cJSON_GetStringValue(backhaul_keypassphrase_item), 128);
+    cJSON_Delete(json);
+    return 0;
+}
+
+static inline int json_parse_backhaul_ssid(char *backhaul_ssid)
+{
+    cJSON *backhaul_ssid_item;
+    cJSON *json = NULL;
+    cJSON_bool valid;
+
+    json = json_open_backhaul_cfg();
+    if (json == NULL) {
+        return -1;
+    }
+
+    backhaul_ssid_item = cJSON_GetObjectItem(json, "Backhaul_SSID");
+    if (!(valid = cJSON_IsString(backhaul_ssid_item))) {
```

Comment:
`cJSON_bool valid;` as far as I can see this variable is not used anywhere.
Consider using this code:
```suggestion

    json = json_open_backhaul_cfg();
    if (json == NULL) {
        return -1;
    }

    backhaul_ssid_item = cJSON_GetObjectItem(json, "Backhaul_SSID");
    if (!cJSON_IsString(backhaul_ssid_item)) {
```